saleor/product/models.py

```diff
@@ -304,6 +309,58 @@ def __str__(self):
         return self.name or str(self.product_variant)
 
 
+class DigitalContent(models.Model):
+    FILE = 'file'
+    TYPE_CHOICES = (
+        (FILE, pgettext_lazy('File as a digital product', 'digital_product')),
+    )
+    automatic_fulfillment = models.BooleanField(default=False)
+    content_type = models.CharField(
+        max_length=128, default=FILE, choices=TYPE_CHOICES)
+    product_variant = models.OneToOneField(
+        ProductVariant, related_name='digital_content',
+        on_delete=models.CASCADE)
+    content_file = models.FileField(upload_to='digital_contents', blank=True)
+    max_downloads = models.IntegerField(blank=True, null=True)
+    url_valid_days = models.IntegerField(blank=True, null=True)
+
+    def create_new_url(self) -> 'DigitalContentUrl':
+        return self.urls.create()
+
+
+class DigitalContentUrl(models.Model):
+    token = models.UUIDField(editable=False, unique=True)
+    content = models.ForeignKey(
+        DigitalContent, related_name='urls', on_delete=models.CASCADE)
+    created = models.DateTimeField(auto_now_add=True)
+    download_num = models.IntegerField(default=0)
+
+    def save(self, force_insert=False, force_update=False, using=None,
+                 update_fields=None):
+        if not self.token:
+            self.token = str(uuid4()).replace('-', '')
+        super().save(force_insert, force_update, using, update_fields)
+
+    def get_absolute_url(self) -> str:
+        url = reverse('product:digital-product', kwargs={'token': str(self.token)})
+        return build_absolute_uri(url)
+
+    def ready_to_share(self) -> bool:
```

Comment:
I'd suggest renaming this function to the short and simple `is_valid`, what do you think?